### PR TITLE
Update SearchView to TypeScript

### PR DESCRIPTION
**Pull request by berkbaski**

### CODE REVIEW

File: ui/searchview/searchview.android.ts [hunk 1]
```diff
@@ -0,0 +1,487 @@
...
+    this.mCompatImageView = this.mSearchEditFrame.getChildAt(0);
+    this.mSearchEditFrame.removeViewAt(0);
+    let a = AndroidConfig.activity.obtainStyledAttributes(null, NativeSupportR.styleable.SearchView, NativeSupportR.attr.searchViewStyle, 0);
+    const mSearchHintIcon = a.getDrawable(NativeSupportR.styleable.SearchView_searchHintIcon); //Drawable
+    this._searchIcon = new Image({ roundedBitmapDrawable: mSearchHintIcon });
+    this.updateQueryHint(this.mSearchSrcTextView, this._searchIcon, this._hint);
+    a.recycle();
+
+    const { ios, android, ...restParams } = params;
+    Object.assign(this.android, this.androidProps);
```
Comment (cenkce) [inline ui/searchview/searchview.android.ts:176]: Object.assign(this._android, this.androidProps, android);

File: ui/searchview/searchview.android.ts [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+    Object.assign(this.ios, ios);
```
Comment (cenkce) [inline ui/searchview/searchview.android.ts:177]: bunu kaldiralim ios'a default deger verelim {} olarak

File: ui/searchview/searchview.ios.ts [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+      this.emit('textChanged', searchText);
+    };
+    this._searchBarDelegate.searchButtonClicked = () => {
+      this._onSearchButtonClicked();
+      this.emit('searchButtonClicked');
+    };
+    this.nativeObject.delegate = this._searchBarDelegate;
+
+    const { ios, android, ...props } = params;
+    Object.assign(this.ios, this.iOSProps);
```
Comment (cenkce) [inline ui/searchview/searchview.ios.ts:157]: Object.assign(this._ios, this.iOSProps);

File: ui/searchview/searchview.ios.ts [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+    Object.assign(this.android, {});
```
Comment (cenkce) [inline ui/searchview/searchview.ios.ts:158]: burayi kaldiralim

**Review by cenkce — CHANGES_REQUESTED** [review]

**Comment by berkbaski** [comment]
I fixed the requests. Could you check it again? @cenkce

**Pull request merged**